ibm/service/cis/resource_ibm_cis_mtls.go

```diff
@@ -0,0 +1,217 @@
+	cisMtlsCertCreatedAt = "created_at"
+	cisMtlsCertUpdatedAt = "updated_at"
+	cisMtlsCertExpireOn  = "expires_on"
+)
+
+func ResourceIBMCISMtls() *schema.Resource {
+	return &schema.Resource{
+		Create:   ResourceIBMCISMtlsCreate,
+		Read:     ResourceIBMCISMtlsRead,
+		Update:   ResourceIBMCISMtlsUpdate,
+		Delete:   ResourceIBMCISMtlsDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisMtlsID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Mtls transaction ID",
+			},
+			cisMtlsCert: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Certificate contents",
+			},
+			cisMtlsCertName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Certificate name",
+			},
+			cisMtlsHostName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Certificate host name",
+			},
+			cisMtlsCertCreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate Created At",
+			},
+			cisMtlsCertUpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate Updated At",
+			},
+			cisMtlsCertExpireOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate Expires on",
+			},
+		},
+	}
+}
+func ResourceIBMCISMtlsCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess)
+	}
+	crn := d.Get(cisID).(string)
+	zoneID := d.Get(cisDomainID).(string)
+	sess.Crn = core.StringPtr(crn)
+
+	options := sess.NewCreateAccessCertificateOptions(zoneID)
+	if name, ok := d.GetOk(cisMtlsCertName); ok {
+		options.SetName(name.(string))
+	}
+
+	if cert_val, ok := d.GetOk(cisMtlsCert); ok {
+		options.SetCertificate(cert_val.(string))
+	}
+
+	if host_val, ok := d.GetOk(cisMtlsHostName); ok {
+		options.SetAssociatedHostnames([]string{host_val.(string)})
+	}
+
+	result, resp, err := sess.CreateAccessCertificate(options)
+	if err != nil || result == nil {
+		return fmt.Errorf("[ERROR] Error creating MTLS access certificate %v", resp)
+	}
+
+	d.SetId(flex.ConvertCisToTfThreeVar(*result.Result.ID, zoneID, crn))
+	return ResourceIBMCISMtlsRead(d, meta)
+
+}
+
+func ResourceIBMCISMtlsRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess)
+	}
+	crn := d.Get(cisID).(string)
```

Comment:
Don't use d.Get in read,update or delete use the d.Id() and get the required varibles
EG  ```
	logpushID, zoneID, crn, _ := flex.ConvertTfToCisThreeVar(d.Id())
	```

Reply:
OK.

Reply:
As i understood, We can use flex.Convert to save value which is not handy during create. But if  can fetch it directly no harm.. Even i agree that way also i can get the  value but i need to make sure it is set during create....